### PR TITLE
Flesh out GET endpoints for OGDS users and teams.

### PR DESCRIPTION
**Pull request by njohner**
For the detail views for teams and ogds users, we need API endpoints. Dummy endoints had already been introduced in https://github.com/4teamwork/opengever.core/pull/6320, now we return the actual data.

### Implementations:
- We refactor the ogds listing endpoints to use summary serlializers (`ISerializeToJsonSummary `) for their items, so that we can reuse them for listing users in the GET of a `team`...
- We add a generic serializer for SQL objects

### Open issues
- API endpoints for SQL objects do not match a view in Plone. This makes it impossible to switch from classic UI to new frontend.
- These are not plone objects and therefore lack other things in the response like breadcrumbs, actions, etc...

A solution to these is proposed in https://github.com/4teamwork/opengever.core/pull/6338

For https://github.com/4teamwork/gever-ui/issues/967 and https://github.com/4teamwork/gever-ui/issues/966

Definition of Done: https://github.com/4teamwork/opengever.core/wiki/Definition-of-Done

## Checkliste
- [x] Changelog-Eintrag vorhanden/nötig?
- [x] Aktualisierung Dokumentation vorhanden/nötig?

### CODE REVIEW

File: opengever/api/ogdslistingbase.py [hunk 1]
```diff
@@ -128,3 +124,10 @@ def extend_query_with_batching(self, query, b_start, b_size):
         query = query.offset(b_start)
         query = query.limit(b_size)
         return query
+
+    @property
+    def item_base_get_url(self):
+        """URL used to construct the GET url for items in the listing,
+        of the form item_base_get_url/endpoint_name/item_id
+        """
+        return self.context.absolute_url()
```
Comment (elioschmutz) [inline opengever/api/ogdslistingbase.py]: Wird anscheinend nirgends gebrauch... ✂️

File: opengever/api/ogdsuser.py [hunk 2]
```diff
@@ -23,11 +27,31 @@ def publishTraverse(self, request, name):
...
+        data['teams'] = []
+        for group in groups:
+            group_serializer = queryMultiAdapter(
+                (group, self.request), ISerializeToJsonSummary)
+            data['groups'].append(group_serializer())
+            for team in group.teams:
+                team_serializer = queryMultiAdapter(
+                    (team, self.request), ISerializeToJsonSummary)
+                data['teams'].append(team_serializer())
+        return data
```
Comment (elioschmutz) [inline opengever/api/ogdsuser.py]: Sollten die Relationen nicht bereits im User-Serializer berücksichtigt werden?

File: opengever/api/team.py [hunk 3]
```diff
@@ -22,12 +26,27 @@ def publishTraverse(self, request, name):
...
+        data.update(serializer())
+
+        # We add the team members
+        data['users'] = []
+        for user in team.group.users:
+            user_serializer = queryMultiAdapter(
+                (user, self.request), ISerializeToJsonSummary)
+            data['users'].append(user_serializer())
+
+        return data
```
Comment (elioschmutz) [inline opengever/api/team.py]: Das gleiche hier: sollten die Referenzen nicht teil des Team-Serializers sein?

File: docs/public/dev-manual/api/users.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
+.. _users:
+
+Benutzer
+========
+
+Benutzer Daten werden mit dem ``@ogds-user`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und  erwartet als Pfad-Argument die ID des Benutzers und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
```
Comment (elioschmutz) [inline docs/public/dev-manual/api/users.rst]:
```suggestion
Benutzer Daten werden mit dem ``@ogds-user`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und erwartet als Pfad-Argument die ID des Benutzers und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
```

File: docs/public/dev-manual/api/users.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        ],
+        "userid": "peter.mueller",
+        "zip_code": null
+      }
+
+
+Teams
+=====
+
+Team Daten werden mit dem ``@team`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und  erwartet als Pfad-Argument die ID des Teams und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
```
Comment (elioschmutz) [inline docs/public/dev-manual/api/users.rst]:
```suggestion
Team Daten werden mit dem ``@team`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und erwartet als Pfad-Argument die ID des Teams und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
```

File: docs/public/dev-manual/api/users.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Benutzer
+========
+
+Benutzer Daten werden mit dem ``@ogds-user`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und  erwartet als Pfad-Argument die ID des Benutzers und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
+
+``http://example.org/fd/kontakte/@ogds-user/peter.mueller``
+
+Weil ein Benutzer nicht ein Plone Inhalt ist, sondern nur einen Eintrag in der
+SQL Datenbank, beinhaltet die Response weniger Information als für andere
+Inhaltstypen.
```
Comment (elioschmutz) [inline docs/public/dev-manual/api/users.rst]:
```suggestion
Ein Benutzer wird lediglich durch einen Eintrag in der SQL Datenbank repräsentiert und ist kein Plone Inhaltstyp. Deshalb beinhaltet die Response weniger Information als für andere Inhaltstypen.
```

File: docs/public/dev-manual/api/users.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+Teams
+=====
+
+Team Daten werden mit dem ``@team`` Endpoint abgefragt. Dieser Endpoint unterstützt nur das GET und  erwartet als Pfad-Argument die ID des Teams und wird auf dem Kontakt Folder abgefragt. Die URL setzt sich somit folgendermassen zusammen:
+
+``http://example.org/fd/kontakte/@team/90``
+
+Weil ein Team nicht ein Plone Inhalt ist, sondern nur einen Eintrag in der
+SQL Datenbank, beinhaltet die Response weniger Information als für andere
+Inhaltstypen.
```
Comment (elioschmutz) [inline docs/public/dev-manual/api/users.rst]:
```suggestion
```suggestion
Ein Team wird lediglich durch einen Eintrag in der SQL Datenbank repräsentiert und ist kein Plone Inhaltstyp. Deshalb beinhaltet die Response weniger Information als für andere Inhaltstypen.
```
```

**Comment by njohner** [comment]
Also see https://4teamwork.slack.com/archives/C060FEYKA/p1584704367037900

File: opengever/api/ogdslistingbase.py (hunk 1 above)
Comment (njohner) [reply opengever/api/ogdslistingbase.py]: Oh sorry, leftovers from the first implementation I've made without serializers

File: opengever/api/ogdsuser.py (hunk 2 above)
Comment (njohner) [reply opengever/api/ogdsuser.py]: Hum, I'm not too sure. `group`s are a relation on the `User` model but not `team`s. On the other hand having everything in the serializer does make sense. That's how we want to represent the object.

Comment (elioschmutz) [reply opengever/api/ogdsuser.py]: I'm not sure neither. So if you think it's ok, then 👍

Comment (njohner) [reply opengever/api/ogdsuser.py]: Ok I moved everything to the serializer.

**Comment by njohner** [comment]
@elioschmutz I included all your suggestions, notably moved adding the serialized related items (groups, members and such) from the endpoint into the corresponding serializers.

I also made some small changes for the new frontend (last commit). This is to also return the `org_unit_title` for teams as well as the group title. There instead of adding a new entry for the group title, I instead add a `group` entry containing the summary serialization of the group. This makes more sense as this is also a related item and will allow the new frontend to also link to the group for example.

**Pull request merged**